### ISSUE TITLE
Improve release generator

### ISSUE THREAD

**Issue by TimvdLippe** [mockito/mockito#478]
Currently releases are generated automatically and documentation pushed to the website. However, when we are publishing a major release, this should be changed.

Probably the best solution is to disable the automatic generation for now and invoke this manually. This does require some changes to the current Gradle release process.

Also, in #79 the changelog was improved. We should take a look at improving this as well.

TLDR: Take a good look at the Gradle release process and purge/improve tasks.

**Comment by bric3** [comment]
I don't think so, I believe we should keep up-to-date javadoc, but maybe pointing to released versions javadocs.

**Comment by bric3** [comment]
Or we could tweak the layout :

* `current` == latest released version => `http://site.mockito.org/mockito/docs/current/org/mockito/Mockito.html`
* `release/2.0.0` => `http://site.mockito.org/mockito/docs/2.0.0/org/mockito/Mockito.html`
* `beta/3.0.0.beta.xxxx` => `http://site.mockito.org/mockito/docs/beta/3.0.0.beta.xxxx/org/mockito/Mockito.html`

We could have some way with javascript injection to have some menu that help switch to other versions.

Anyway I think we could safely purge beta javadoc publication once 2.0.0 release https://github.com/mockito/mockito/tree/gh-pages/docs

**Comment by TimvdLippe** [comment]
This issue description is absolutely horrible when I read it back now. It was mainly concerning the version issues, which should be fixed in #483. The other issue is indeed the documentation, but this is mainly an issue for the website, no?

Anyways, it would be to merge #483 sooner rather than later as Mockito 2.0 has been lingering for a while now. The longer we wait, the less incentivized will people be.

**Comment by TimvdLippe** [comment]
This issue has been resolved with the new version scheme and possibility to manually release a version
